qa/rpc-tests/walletbackup.py

```diff
@@ -191,6 +191,30 @@ def run_test(self):
         self.start_three()
         sync_blocks(self.nodes)
 
+        # We made extra transactions that involved addresses generated after the
```

Comment:
Is this situation one that users will encounter?

Reply:
Almost definitely. This test was checking the scenario where a user makes a backup, then creates some transactions, and then loses their wallet. Previously what saved the user was the keypool: as long as they didn't make too many transactions, they would be using pre-generated keys that were included in the backup. Now, what saves them is that keys are HD, so the backup will produce all future keys.

The fix to the counter was in fact necessary to get this test to pass, because the precise sequence of change addresses affected the sequence of `getnewaddress` outputs (this is non-deterministic due to how `CreateTransaction` from upstream works). By splitting the counters, `getnewaddress` is now entirely deterministic for a given mnemonic.

Reply:
Okay, I understand the situation better now. Since we don't yet implement restore of the mnemonic from backup anyway, this will get resolved when that is implemented.